### PR TITLE
Implement Prometheus client

### PR DESCRIPTION
**Pull request by tkulczynski**

### CODE REVIEW

**Comment by tkulczynski** [comment]
cc: @kgrygiel, @mwielgus

File: vertical-pod-autoscaler/recommender/recommender.go [hunk 1]
```diff
@@ -35,13 +39,59 @@ type Recommender interface {
 }
 
 type recommender struct {
+	clusterState            model.ClusterState
 	specClient              cluster.SpecClient
 	metricsFetchingInterval time.Duration
+	prometheusClient        signals.PrometheusClient
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/recommender.go]: can we have an interface that will hide prometheus details from recommender?

File: vertical-pod-autoscaler/recommender/signals/prometheus_util.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import (
+	"fmt"
+	"io"
+	"encoding/json"
+	"time"
+)
+
+// Helper types used for parsing json returned by Prometheus.
+// This is the top-level structure of the response.
+type responseType struct {
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/signals/prometheus_util.go]: Add an information that prometheus didn't make this api available in form of any go library. 
https://github.com/prometheus/prometheus/blob/2d73d2b892853e95dbf157561e9df56ac220875e/web/api/v1/api.go#L92

File: vertical-pod-autoscaler/recommender/signals/prometheus_client.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+package signals
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// A client that talks to Prometheus using its HTTP API.
+type PrometheusClient interface {
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/signals/prometheus_client.go]: Why are we not using GRPC v2 api?

File: vertical-pod-autoscaler/recommender/recommender.go (hunk 1 above)
Comment (tkulczynski) [reply vertical-pod-autoscaler/recommender/recommender.go]: Left a TODO for now, there's plenty of work and I do not want to have a single PR that makes world perfect.

File: vertical-pod-autoscaler/recommender/signals/prometheus_client.go (hunk 3 above)
Comment (tkulczynski) [reply vertical-pod-autoscaler/recommender/signals/prometheus_client.go]: It's not listed on the official docs: https://prometheus.io/docs/prometheus/latest/querying/api/. I also couldn't find much information about it, other than the implementation on github, so I'm not sure what state it's in (presumably under development).

**Comment by mwielgus** [comment]
Merging manually. Tests passed.

**Pull request merged**